test/cli_test.rb

```diff
@@ -0,0 +1,67 @@
+require "test_helper"
+require "date"
+
+class GemDating::CliTest < Minitest::Test
+  def setup
+    @spec1 = Gem::Specification.new do |s|
+      s.name = "rest-client"
+      s.version = "2.1.0"
+      s.date = DateTime.parse("2019-08-21")
+    end
+    @spec2 = Gem::Specification.new do |s|
+      s.name = "rails"
+      s.version = "7.0.5"
+      s.date = DateTime.parse("2023-05-24")
+    end
+    @spec3 = Gem::Specification.new do |s|
+      s.name = "graphql"
+      s.version = "2.0.22"
+      s.date = DateTime.parse("2023-05-17")
+    end
+  end
+
+  def test_gemfile
+    exit_code = nil
+
+    rubygems = Mocktail.of(GemDating::Rubygems)
+
+    stubs { |m| rubygems.fetch(m.is_a?(Array)) }.with { [@spec1, @spec2, @spec3] }
```

Comment:
I'm not sure this is actually working as intended.  It turns out graphql dropped a new release yesterday and the specs are failing for me:

```
GemDating::CliTest#test_gemfile [/Users/stevejackson/testdouble/gem_dating/test/cli_test.rb:43]:
--- expected
+++ actual
@@ -2,5 +2,5 @@
 ------------|---------|-----------
 rest-client | 2.1.0   | 2019-08-21
 rails       | 7.0.5   | 2023-05-24
-graphql     | 2.0.22  | 2023-05-17
+graphql     | 2.0.23  | 2023-06-19
 ```
This may be confusing because I created both an instance and class method called `fetch`.   `of_next` might get what you want, but happy to shuffle things around if the names are just bad here.

Reply:
Great catch here, thank you!

Besides being a good reminder for using highly sus test data, poking at Mocktail more was nice.

I wound up swapping out the entire `Rubygems` class, to return the expected Gemspecs. 

Mocktail is pretty suspicious of both class & instance methods being given the same name, if we lean on `of_next`:

```
Mocktail::AmbiguousDemonstrationError: `stubs` & `verify` expect exactly one invocation of a mocked method, but 2 were detected.
```

I think the way you exposed the results makes sense: we don't need the instance of `Rubygems`. I'd be curious what the magic incantation is for Mocktail to gracefully accept a call to the class level `fetch`, and politely mock out the next instance that gets returned on `new`, but that's not really needed for our case here.